src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -39,13 +44,16 @@ void create_pruning_statistics_for_segment(AttributeStatistics<T>& segment_stati
 }  // namespace
 
 namespace hyrise {
+bool pruning_statistics_should_be_generated_for_chunk(const std::shared_ptr<Chunk>& chunk) {
+  // We do not generate statistics for chunks as long as they are mutable.
+  // Also, pruning statistics should be stable no matter what encoding or sort order is used.
+  // Hence, when they are present they are up to date, and we can skip the recreation.
+  return chunk && !chunk->is_mutable() && !chunk->pruning_statistics();
+}
 
 void generate_chunk_pruning_statistics(const std::shared_ptr<Chunk>& chunk) {
-  if (chunk->pruning_statistics()) {
-    // Pruning statistics should be stable no matter what encoding or sort order is used. Hence, when they are present
-    // they are up to date and we can skip the recreation.
-    return;
-  }
+  DebugAssert(pruning_statistics_should_be_generated_for_chunk(chunk),
+              "Method should only be called for chunks that need it.");
```

Comment:
Well, no chunk really "needs" it. Maybe the optimizer "needs" it. I think "qualifying chunks" might be enough as the function above is very clear.